src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/RazorFileChangeDetector.cs

```diff
@@ -44,8 +47,21 @@ internal class RazorFileChangeDetector : IFileChangeDetector
             _filePathNormalizer = filePathNormalizer;
             _listeners = listeners;
             _watchers = new List<FileSystemWatcher>(RazorFileExtensions.Count);
+            _pendingNotifications = new Dictionary<string, DelayedFileChangeNotification>(FilePathComparer.Instance);
```

Comment:
Just a nitpick, but if this doesn't use any constructor parameters why not assign it up on line 23? I guess the same can be said of `_watchers`, I just feel like it keeps the constructor a bit cleaner if anything that doesn't use it's parameters is instantiated inline.

Reply:
I don't have a hard fast rule for where to instantiate read-only variables but I've just gotten used to assigning statics in-line and everything else in the ctor.